inc/Engine/Preload/Database/Queries/Cache.php

```diff
@@ -289,7 +291,7 @@ public function get_old_cache() : array {
 		}
 
 		$prefixed_table_name = $db->prefix . $this->table_name;
-		$query               = "SELECT id FROM `$prefixed_table_name` WHERE `last_accessed` <= date_sub(now(), interval 1 month)";
+		$query               = "SELECT id FROM `$prefixed_table_name` WHERE `last_accessed` <= date_sub(now(), interval $delay)";
```

Comment:
why not use the filter here also instead of adding a parameter to the method?

Reply:
@Tabrisrp is some previous CR @engahmeds3ed told me to not add filters inside filters inside the Query.

Reply:
I don't understand why it would be different from the current implementation 🤔

Reply:
You have an example of current implementation [here](https://github.com/wp-media/wp-rocket/blob/ef300cf2d46e837789f9ad42ff8166c832a47dc3/inc/Engine/Preload/Database/Queries/Cache.php#L666) where the filter is declared [here](https://github.com/wp-media/wp-rocket/blob/ef300cf2d46e837789f9ad42ff8166c832a47dc3/inc/Engine/Preload/Controller/PreloadUrl.php#L191).

Reply:
```suggestion
               $delay = (string) apply_filters( 'rocket_preload_delay_delete_non_accessed', '1 month' );
		$query               = "SELECT id FROM `$prefixed_table_name` WHERE `last_accessed` <= date_sub(now(), interval $delay)";
```

I still don't understand why we can't do that?

Reply:
if we do it like that we need to find a different way to pass the value to the parameter, a string like this is not great

Reply:
@Tabrisrp I changed the method params to adapt to it.